### PR TITLE
Yaml search through deployable directory

### PR DESCRIPTION
**Pull request by loosebazooka**
@meltsufin open to suggestions on different ways to do this, basically if we get a directory, then look through it for yamls

### CODE REVIEW

File: src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineDeployment.java [hunk 1]
```diff
@@ -48,12 +53,22 @@ public void deploy(DeployConfiguration config) throws AppEngineException {
...
     List<String> arguments = new ArrayList<>();
     arguments.add("deploy");
-    for (File deployable : config.getDeployables()) {
-      if (!deployable.exists()) {
-        throw new IllegalArgumentException(
-            "Deployable " + deployable.toPath().toString() + " does not exist.");
+    // If we get a single directory, then look in it for yamls
+    if (config.getDeployables().size() == 1 && config.getDeployables().get(0).isDirectory()) {
+      for (String filename : yamlFilenames) {
+        File yamlFile = new File(config.getDeployables().get(0), filename);
```
Comment (meltsufin) [inline src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineDeployment.java]: I would put `config.getDeployables().get(0)` into a local variable so you don't have to repeat it.

**Comment by meltsufin** [comment]
I like that this logic is in the common lib, but I'm unsure about allowing a directory to be passed in as a "deployable" and that having a special meaning.

File: src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineDeployment.java [hunk 2]
```diff
@@ -24,14 +24,20 @@
...
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 
 /**
  * Cloud SDK based implementation of {@link AppEngineDeployment}.
  */
 public class CloudSdkAppEngineDeployment implements AppEngineDeployment {
 
+  private static final List<String> yamlFilenames = Arrays
```
Comment (meltsufin) [inline src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineDeployment.java]: Can you make the variable all upper case? Like APPENGINE_YAML_FILES?

Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineDeployment.java]: good call.

**Comment by meltsufin** [comment]
:ship:

**Pull request merged**